### PR TITLE
TL-56: Add Activity History Shared Code

### PR DESCRIPTION
**Pull request by notatiyyah**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/TL-56

## Describe this PR

- Add the ActivityHistory domain objects and tests.
- Add the ActivityHistory request & response objects (+ validation)
- Add the ActivityHistory database entity
- Add factories
- Update the pipeline to run the tests and check code formatting for each push.

#### _Checklist_

- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Update the Activity History API and Activity Listener to use this package.

### CODE REVIEW

**Review by evak6979 — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -30,12 +33,44 @@ jobs:
...
+    needs: calculate-version
+    env:
+      LBHPACKAGESTOKEN: ${{secrets.LBHPACKAGESTOKEN }}
+    outputs:
+      version: ${{ needs.calculate-version.outputs.version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build
+        run: docker-compose build hackney-shared-activity-test
```
Comment (humulla) [inline .github/workflows/publish.yml:63]: Can we please call it activity-history to keep things consistent

**Review by humulla — APPROVED** [review]
Small change, but the rest looks good! :)

File: .github/workflows/publish.yml (hunk 1 above)
Comment (notatiyyah) [reply .github/workflows/publish.yml:63]: The name of the whole package?

Comment (humulla) [reply .github/workflows/publish.yml:63]: Whole package if possible and docker image as well please

Comment (notatiyyah) [reply .github/workflows/publish.yml:63]: I'm happy to change it, but according to this ticket: https://hackney.atlassian.net/browse/MTTL-1117, it should be Activity Shared.

Comment (humulla) [reply .github/workflows/publish.yml:63]: I've asked Mirela and Micheal in the relative ticket

**Review by humulla — APPROVED** [review]
LGTM

**Pull request merged**